controllers/operator_controller.go

```diff
@@ -162,14 +162,35 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		return ctrl.Result{}, err
 	}
 
-	// update operator status
+	// convert existing unstructured object into bundleDeployment for easier mapping of status.
+	existingTypedBundleDeployment := &rukpakv1alpha1.BundleDeployment{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(dep.UnstructuredContent(), existingTypedBundleDeployment); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update operator status:
+	// 1. If the Operator "Ready" status is "Unknown": The status of successful bundleDeployment is unknown, requeue the request.
+	// 2. If the Operator "Ready" status is "True": Update the "successful resolution" status and return the result.
+	// 3. If the Operator "Ready" status is "False": There is error observed from Rukpak. Update the status accordingly.
+	status, message := verifyBDStatus(existingTypedBundleDeployment)
+	var reason string
+	if status == metav1.ConditionTrue || status == metav1.ConditionUnknown {
+		reason = operatorsv1alpha1.ReasonResolutionSucceeded
```

Comment:
I'm thinking `ReasonResolutionSucceeded` is no longer the right thing. We might need to rename this to something like `ReasonInstallSucceeded`

Reply:
+1! Not sure if we want to do in this PR, but I've created a separate [commit](https://github.com/operator-framework/operator-controller/pull/125/commits/543b7a4d26d21d5ff8a800884ca3b6917c6def7d) with the change of `ReasonResolutionSucceeded` to `ReasonInstallationSucceeded`. If we want to remove it out of this PR, I can create a follow-up. We could make another iteration of re-naming condition type and reasons for addressing https://github.com/operator-framework/operator-controller/issues/126 in a follow-up.